docs/sources/source/training.rst

```diff
@@ -22,13 +22,11 @@ Multi-GPU Training
 
 For multi-GPU training:
 
-(1) Set `placement` to `nemo.core.DeviceType.AllGpu` in NeuralModuleFactory
-(2) Add 'local_rank' argument to your script and do not set it yourself: `parser.add_argument("--local_rank", default=None, type=int)`
+Add 'local_rank' argument to your script and do not set it yourself: `parser.add_argument("--local_rank", default=os.getenv('LOCAL_RANK',0), type=int)`
```

Comment:
No, this is wrong.
Default should be None, otherwise the code in Neural Factory will break. Especially if you choose not to pass placement to NeuralFactory anymore.

Reply:
Why would it break? 
If torch.distributed launcher was used then --local_rank argument will be passed in command line
If slurm is used, then LOCAL_RANK  env var will be set
Otherwise it will be 0

Reply:
If  --local_rank argument is not passed AND slurm is not used, then it will default to 0.
If the defaults to 0 and you do not pass placement to NeuralFactory, NeuralFactory will default to ALLGPU which is not what you want for single GPU cases.

Advising people to add this to their scripts even if it is under the multi-gpu section is bad advice. We shouldn't do it.

Reply:
Changed to os.getenv('LOCAL_RANK', None)